### PR TITLE
added support for CA2241

### PR DESCRIPTION
**Pull request by heejaechang**
CA2241 is "Provide correct arguments to formatting methods"
(https://msdn.microsoft.com/en-us/library/ms182361.aspx)

due to some bugs in roslyn, some cases are not working correctly. one is __arglist support in C# and the other is explicit params array arguments in C#.

right now, this analyzer will pass on analyzing those two cases.

### CODE REVIEW

**Comment by heejaechang** [comment]
@srivatsn @mavasani @JohnHamby @lgolding @natidea @jinujoseph @basoundr can you take a look? thank you

**Comment by mavasani** [comment]
I thought @jmarolf already implemented this analyzer a while back, or was it some other rule? He should probably be the best person to review this.

**Comment by heejaechang** [comment]
@mavasani @jmarolf ah, didn't know. it was not in roslyn-analyzer though.

**Comment by heejaechang** [comment]
ping?

File: src/System.Runtime.Analyzers/Core/ProvideCorrectArgumentsToFormattingMethods.cs [hunk 1]
```diff
@@ -4,36 +4,421 @@
...
+                    var actualArgumentCount = dimensionInitializer.ElementValues.Length;
+                    if (actualArgumentCount != expectedStringFormatArgumentCount)
+                    {
+                        operationContext.ReportDiagnostic(operationContext.Operation.Syntax.CreateDiagnostic(Rule));
+                    }
+                }, OperationKind.InvocationExpression);
+            });
+        }
+
+        private int GetFormattingArguments(string format)
```
Comment (srivatsn) [inline src/System.Runtime.Analyzers/Core/ProvideCorrectArgumentsToFormattingMethods.cs]: Was this code picked up from the corefx repo's implementation of string.Format? If so, can you add a comment\link to that?

**Comment by srivatsn** [comment]
I think @jmarolf was going to implement this rule but didn't.

**Comment by srivatsn** [comment]
LGTM

**Comment by srivatsn** [comment]
Oh can update the props file as well for this rule?

File: src/System.Runtime.Analyzers/CSharp/System.Runtime.CSharp.Analyzers.props [hunk 2]
```diff
@@ -10,6 +10,7 @@
       <CodeAnalysisRuleSetOverrides>
          $(CodeAnalysisRuleSetOverrides);
          -Microsoft.Performance#CA1820;
+         -Microsoft.Usage#CA2241;
```
Comment (heejaechang) [inline src/System.Runtime.Analyzers/CSharp/System.Runtime.CSharp.Analyzers.props]: I already did

Comment (srivatsn) [inline src/System.Runtime.Analyzers/CSharp/System.Runtime.CSharp.Analyzers.props]: Missed that. :+1:

**Pull request merged**